source/WB-CAN-Firmware.c

```diff
@@ -166,28 +201,90 @@ int main(void) {
     for(;;);
 }
 
-/*******************************************************************************
- * Tasks
- ******************************************************************************/
-static void blinkTask(void *pv) {
-	while(1) {
 
-		vTaskDelay(pdMS_TO_TICKS(500));
-//		printf("BLINK");
+// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
+// Main Tasks:
+// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
+
+
+// Initialize Semaphores:
+SemaphoreHandle_t semaphore_Message;
+SemaphoreHandle_t semaphore_PWMActive;
+
+// Create Message Type that takes specific values using Enumerate:
+enum messageEnum
+    {
+        No_Command, DPR_Pause, DPR_Resume
+    };
+typedef enum messageEnum messageType;
+messageType message;
+
+/*
+// Description:	This task reads the current message and sets the program to different settings.
+// Note:		This is not a queue and any message written twice before being acted on is overwritten.
+//				Future Revision will likely be a queue.
+*/
+static void mainTask(void *pv){
+
+	// Create Semaphores:
+    semaphore_Message = xSemaphoreCreateBinary();
+    semaphore_PWMActive = xSemaphoreCreateBinary();
+
+
+    // Forever Loop:
+	for(;;){
+		vTaskDelay(pdMS_TO_TICKS(200));
+
+		// Wait for message
+		while(uxSemaphoreGetCount(semaphore_Message) == 0){
+		};
+
+		// Semaphore is Available, so now we can take it and read the message:
+		xSemaphoreTake(semaphore_Message, 10);
+
+		// Feature Setting Logic:
+		if (message == DPR_Pause){
+			xSemaphoreTake( semaphore_PWMActive, 10 );
+			printf("Command Executed: DPR_Pause.\n");
+		}
+		else if(message == DPR_Resume){
+			xSemaphoreGive(semaphore_PWMActive);
+			printf("Command Executed: DPR_Resume.\n");
+		}
+		else {
+			//pass;
+		}
```

Comment:
```suggestion
		switch(message){
			case DPR_Pause:
				xSemaphoreTake( semaphore_PWMActive, 10 );
				printf("Command Executed: DPR_Pause.\n");
				break;
			case DPR_Resume:
				xSemaphoreGive(semaphore_PWMActive);
				printf("Command Executed: DPR_Resume.\n");
				break;
		}
```

The really cool thing about switch used with an enum is that most compilers (I think our GCC as well) will throw a warning if all entries in the enum are not covered by the switch. Plus, I at least think it's a bit cleaner. I wrote this in a browser, so please check the syntax and formatting though.